web/src/pages/user-profile-edit.js

```diff
@@ -0,0 +1,163 @@
+import AuthService from '../utils/auth-service'
+import FormErrors from '../components/form-errors'
+import React, { useEffect, useState } from 'react'
+import useAuth from '../hooks/use-auth'
+import {
+  Button,
+  Card,
+  CardActions,
+  CardContent,
+  Container,
+  FormControl,
+  Grid,
+  TextField,
+  Typography,
+} from '@material-ui/core'
+import { Formik } from 'formik'
+import { makeStyles } from '@material-ui/core/styles'
+import { useRouter } from 'next/router'
+
+const useStyles = makeStyles({
+  root: {
+    maxWidth: 500,
+    margin: '30px auto 0',
+  },
+  title: {
+    fontSize: 18,
+  },
+  formInputs: {
+    width: '100%',
+    marginBottom: 15,
+  },
+  fullWidth: {
+    display: 'block',
+    width: '100%',
+    textAlign: 'center',
+  },
+})
+
+const UserProfileEdit = () => {
+  const classes = useStyles()
+  const [formErrors, setFormErrors] = useState([])
+  const [isLoading, setIsLoading] = useState(true)
+  const [formValues, setFormValues] = useState({
+    id: null,
+    user_name: '',
+    photo_url: '',
+  })
+  const router = useRouter()
+  const { isLoggedIn, getUserProfile, updateUserProfile } = useAuth()
+
+  useEffect(() => {
+    if (isLoggedIn){
+      // If user is logged in, show loading & fetch their profile data
+      console.log('get profile info')
+      getUserProfile().then( (res) => {
+        console.log('getUserProfile res', res)
+        setIsLoading(false)
+      }).catch( (err) => {
+        console.log('getUserProfile err', err)
+        setIsLoading(false)
+      })
+    } else {
+      // If user is logged out, kick them to the sign-in
+      router.push('/login')
+    }
+
+  }, [isLoggedIn, router, getUserProfile])
+
+  if(isLoading){
+    return (
+      <span>Loading...</span>
+    )
+  }
+
+  return (
+    <Container>
+      <Card className={classes.root}>
+        <Formik
+          initialValues={formValues}
+          onSubmit={(formData, { setSubmitting }) => {
+            const newProfile = !!formData.id
```

Comment:
The boolean here is what is causing the issue. This currently is returning true when there is an id in the formData (which in turn is causing it to be treated as a new profile, causing a post and not a put). Let's switch it to `!formData.id`. In the case it is not defined it should just be null since it is parsing from JSON.